coreclient/src/clients/add_contact.rs

```diff
@@ -39,6 +43,81 @@ use super::{
 };
 
 impl CoreUser {
+    /// Create a connection with via a user handle.
+    pub(crate) async fn add_contact_via_handle(
+        &self,
+        handle: UserHandle,
+    ) -> anyhow::Result<ConversationId> {
+        let client = self.api_client()?;
+
+        // Phase 1: Fetch a connection package from the AS
+        let (connection_package, connection_offer_responder) =
+            client.as_connect_handle(&handle).await?;
+
+        // Phase 2: Verify the connection package
+        let as_intermediate_credential = AsCredentials::get(
+            self.pool().acquire().await?.as_mut(),
+            &self.inner.api_clients,
+            self.user_id().domain(),
+            connection_package.client_credential_signer_fingerprint(),
+        )
+        .await?;
+        let verifying_key = as_intermediate_credential.verifying_key();
+        let connection_package = connection_package.verify(verifying_key)?;
+
+        // Phase 3: Create a connection group
+        let group_id = client.ds_request_group_id().await?;
+
+        // Phase 4: Prepare the connection locally
+        let mut connection = self.pool().acquire().await?;
+        let mut notifier = self.store_notifier();
+        let connection_package = VerifiedConnectionPackagesWithGroupId {
+            verified_connection_packages: vec![connection_package],
+            group_id,
+        };
+
+        let contact_id = ContactId::Handle(handle);
+
+        let local_group = connection
+            .with_transaction(async |txn| {
+                connection_package
+                    .create_local_connection_group(
+                        txn,
+                        &mut notifier,
+                        &self.inner.key_store,
+                        contact_id.clone(),
+                    )
+                    .await
+            })
+            .await?;
+
+        let client_reference = self.create_own_client_reference();
+
+        let local_partial_contact = local_group
+            .create_partial_contact(
+                &mut connection,
+                &mut notifier,
+                &self.inner.key_store,
+                client_reference,
+                self.user_id(),
+                contact_id,
+            )
+            .await?;
```

Comment:
What happens when the client goes offline here? I guess it leaves us with an orphaned connection group. Can we clean this up later?